### PR TITLE
Added support to configure date time format from settings page and updated UI of settings page(#2t2y3qq)

### LINKED ISSUE #87: Update the UI of settings page and add support to configure date time format
## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by disha1202**
Closes #87 

Screenshot from 2022-09-19 13-50-21

Screenshot from 2022-09-19 13-38-23

Screenshot from 2022-09-27 12-04-20

### CODE REVIEW

File: src/components/DateTimeModal.vue [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  IonToolbar,
+  modalController,
+  alertController } from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, save } from "ionicons/icons";
+import { useStore } from "@/store";
+import { UserService } from "@/services/UserService";
+import { hasError } from '@/utils'
+import { DateTime } from 'luxon';
+import { mapGetters } from "vuex";
```
Comment (ymaheshwari1) [inline src/components/DateTimeModal.vue:55]: Use a single import for vuex

File: src/components/DateTimeModal.vue [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    
+    selectSearchBarText(event: any) {
+      event.target.getInputElement().then((element: any) => {
+        element.select();
+      })
+    },
+    
+  },
+  mounted(){
+      this.format = this.dateTimeFormat;
```
Comment (ymaheshwari1) [inline src/components/DateTimeModal.vue:97]: Improve indentation.

File: src/views/Settings.vue [hunk 3]
```diff
@@ -8,40 +8,158 @@
     </ion-header>
     
     <ion-content :fullscreen="true">
+      <ion-card class="user-profile">
+        <ion-item lines="full">
+          <ion-thumbnail>
```
Comment (dt2patel) [inline src/views/Settings.vue:13]: ion-avatar, missing slot

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue [hunk 4]
```diff
@@ -8,40 +8,158 @@
...
+            <img src="https://dev-resources.hotwax.io/resources/uploads/images/product/m/h/mh09-blue_main.jpg" />
+          </ion-thumbnail>
+          <ion-label>
+            First Last
+            <p>username</p>
+          </ion-label>
+        </ion-item>
+        <ion-button size="medium" fill="outline" color="danger" @click="logout()">{{ $t("Logout") }}</ion-button>
```
Comment (dt2patel) [inline src/views/Settings.vue]: do you need to define size?

File: src/views/Settings.vue [hunk 5]
```diff
@@ -8,40 +8,158 @@
...
+        <ion-button size="medium" fill="outline" >{{ $t("Reset password") }}</ion-button>
```
Comment (dt2patel) [inline src/views/Settings.vue]: missing color

File: src/views/Settings.vue [hunk 6]
```diff
@@ -8,40 +8,158 @@
...
+      </ion-card>
       <!-- OMS information -->
       <ion-item>
```
Comment (dt2patel) [inline src/views/Settings.vue:25]: all cards should not be in an ion item

File: src/views/Settings.vue [hunk 7]
```diff
@@ -8,40 +8,158 @@
...
-        <ion-button @click="changeTimeZone()" slot="end" fill="outline" color="dark">{{ $t("Change") }}</ion-button>
-      </ion-item>
-      <!-- Profile of user logged in -->
-      <ion-item>
-        <ion-icon :icon="personCircleOutline" slot="start" />
-        <ion-label>{{ userProfile !== null ? userProfile.partyName : '' }}</ion-label>
-        <ion-button slot="end" fill="outline" color="dark" @click="logout()">{{ $t("Logout") }}</ion-button>
-      </ion-item>
+        <div>
+          <ion-label>{{ $t('OMS') }}</ion-label>
```
Comment (dt2patel) [inline src/views/Settings.vue:27]: check Figma and see which HTML tag should be used.

File: src/views/Settings.vue [hunk 8]
```diff
@@ -8,40 +8,158 @@
...
+              </ion-button>
+            </ion-card>
+  
+            <ion-card>
+              <ion-card-header>
+                <ion-card-subtitle>
+                  <p>{{ $t("Product store")}}</p>
+                </ion-card-subtitle>
+                <ion-card-title>
+                  Store
```
Comment (dt2patel) [inline src/views/Settings.vue:55]: label

File: src/views/Settings.vue [hunk 9]
```diff
@@ -84,10 +215,34 @@ export default defineComponent({
...
+  max-width: 67%;
+}
+.user-profile {
+  max-width: 45%;
+}
+ion-card > ion-button {
+  padding: 0px 6px;
+}
+img {
+  border-radius: 100%;
```
Comment (dt2patel) [inline src/views/Settings.vue:246]: don't need this

File: src/views/Settings.vue [hunk 10]
```diff
@@ -84,10 +215,34 @@ export default defineComponent({
...
+      saveOutline,
       store,
       router,
       timeOutline
     }
   }
 });
 </script>
+<style scoped>
+.oms-info, .app-info{
```
Comment (dt2patel) [inline src/views/Settings.vue:228]: why do you need two classes if they are the same?

Indentation check

File: src/views/Settings.vue [hunk 11]
```diff
@@ -84,10 +215,34 @@ export default defineComponent({
...
+  display: grid;
+  grid-template-columns: auto auto auto;
```
Comment (dt2patel) [inline src/views/Settings.vue:230]: how will this respond on mobile devices?

File: src/views/Settings.vue [hunk 12]
```diff
@@ -8,40 +8,158 @@
...
+              
+              <ion-card-content>
+                {{ $t('Enter a custom date time format that you want to use when uploading documents to HotWax Commerce.')}}
+              </ion-card-content> 
+              <ion-item>
+                <ion-input clear-input='true' v-on:keyup.enter="dateTimeFormat = $event.target.value; parse()" v-model="dateTimeFormat" :value="dateTimeFormat" />
+              </ion-item>
+              <ion-item>
+                <ion-label>{{ sampleDateTime }}</ion-label>
+                <ion-badge color="warning">sample</ion-badge>
```
Comment (dt2patel) [inline src/views/Settings.vue:133]: label

File: src/views/Settings.vue [hunk 13]
```diff
@@ -8,40 +8,158 @@
...
+                </ion-select>
+              </ion-item>
+            </ion-card>
+          </div>
+        </div>
+      </ion-item>  
+
+      <ion-item lines="none">
+        <div>
+          <ion-label>{{ $t('APP') }}</ion-label>
```
Comment (dt2patel) [inline src/views/Settings.vue:96]: Why all caps?

**Review by dt2patel — CHANGES_REQUESTED** [review]
See comments

File: src/views/Settings.vue [hunk 14]
```diff
@@ -234,20 +234,23 @@ export default defineComponent({
 ion-card > ion-button {
```
Comment (dt2patel) [inline src/views/Settings.vue:234]: still not indented correctly

**Review by dt2patel — CHANGES_REQUESTED** [review]

File: src/store/modules/user/getters.ts [hunk 15]
```diff
@@ -20,7 +20,9 @@ const getters: GetterTree <UserState, RootState> = {
     },
     getInstanceUrl (state) {
         const baseUrl = process.env.VUE_APP_BASE_URL;
-        return baseUrl ? baseUrl : state.instanceUrl;
+        return baseUrl ? baseUrl : state.instanceUrl;    },
```
Comment (ymaheshwari1) [inline src/store/modules/user/getters.ts:23]: Improve formatting

File: src/views/Settings.vue [hunk 16]
```diff
@@ -8,73 +8,210 @@
...
     },
+    parse(){
+      this.sampleDateTime = DateTime.now().toFormat(this.dateTimeFormat);
+    },
     logout () {
       this.store.dispatch('user/logout').then(() => {
         this.router.push('/login');
       })
-    }
+    },
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:214]: Remove unwanted comma separator

File: src/views/Settings.vue [hunk 17]
```diff
@@ -8,73 +8,210 @@
...
+              {{ $t("Date Format") }}
+            </ion-card-title>
+          </ion-card-header>
+          
+          <ion-card-content>
+            {{ $t('Enter a custom date time format that you want to use when uploading documents to HotWax Commerce.') }}
+            <p>{{ $t('Luxon date time formats can be found') }} <a href="https://github.com/moment/luxon/blob/master/docs/formatting.md">here</a></p>
+          </ion-card-content> 
+          <ion-item>
+            <ion-input clear-input='true' v-on:keyup.enter="dateTimeFormat = $event.target.value; parse()" v-model="dateTimeFormat" :value="dateTimeFormat" />
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:127]:
```suggestion
            <ion-input clear-input='true' @keyup.enter="dateTimeFormat = $event.target.value; parse()" v-model="dateTimeFormat" :value="dateTimeFormat" />
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/getters.ts [hunk 18]
```diff
@@ -20,7 +20,10 @@ const getters: GetterTree <UserState, RootState> = {
     },
     getInstanceUrl (state) {
         const baseUrl = process.env.VUE_APP_BASE_URL;
-        return baseUrl ? baseUrl : state.instanceUrl;
+        return baseUrl ? baseUrl : state.instanceUrl;
```
Comment (ymaheshwari1) [inline src/store/modules/user/getters.ts:23]: This change is not needed

File: src/views/Settings.vue [hunk 19]
```diff
@@ -8,68 +8,205 @@
...
+              {{ $t("File upload")}}
+            </ion-card-subtitle>
+            <ion-card-title>
+              {{ $t("Date Format") }}
+            </ion-card-title>
+          </ion-card-header>
+          
+          <ion-card-content>
+            {{ $t('Enter a custom date time format that you want to use when uploading documents to HotWax Commerce.') }}
+            <p>{{ $t('Luxon date time formats can be found') }} <a href="https://github.com/moment/luxon/blob/master/docs/formatting.md">here</a></p>
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:124]: Internationalize text

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue [hunk 20]
```diff
@@ -8,68 +8,168 @@
...
-      <ion-item>
-        <ion-icon :icon="personCircleOutline" slot="start" />
-        <ion-label>{{ userProfile !== null ? userProfile.partyName : '' }}</ion-label>
-        <ion-button slot="end" fill="outline" color="dark" @click="logout()">{{ $t("Logout") }}</ion-button>
-      </ion-item>
+      <div class="user-profile">
+        <ion-card>
+          <ion-item lines="full">
+            <ion-avatar slot="start" v-if="userProfile?.partyImageUrl">
+              <Image :src="userProfile?.partyImageUrl"/>
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:15]: I think optional chaining is not needed as we are already checking partyImageUrl in v-if condition.
```suggestion
              <Image :src="userProfile.partyImageUrl"/>
```

File: src/views/Settings.vue [hunk 21]
```diff
@@ -8,68 +8,168 @@
...
+      currentDateTimeFormat: 'user/getDateTimeFormat'
     })
   },
+  mounted(){
+    this.dateTimeFormat = this.currentDateTimeFormat
+    this.parse();
+  },
   methods: {
+    goToOms(){
+      window.location.href = `https://${this.instanceUrl}.hotwax.io/`;
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:159]: Open the oms in a new tab. Also check whether the instanceUrl contains `http` or not and handle the case accordingly.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue [hunk 22]
```diff
@@ -8,68 +8,168 @@
...
+              {{ $t("File upload")}}
+            </ion-card-subtitle>
+            <ion-card-title>
+              {{ $t("Date Format") }}
+            </ion-card-title>
+          </ion-card-header>
+          
+          <ion-card-content>
+            {{ $t('Enter a custom date time format that you want to use when uploading documents to HotWax Commerce.') }}
+            <p>{{ $t('Luxon date time formats can be found') }} <a href="https://github.com/moment/luxon/blob/master/docs/formatting.md">{{ $t("here") }}</a></p>
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:85]: Change luxon link to `https://moment.github.io/luxon/#/formatting?id=presets`

File: src/store/modules/user/actions.ts [hunk 23]
```diff
@@ -18,6 +18,7 @@ const actions: ActionTree<UserState, RootState> = {
       const resp = await UserService.login(username, password)
       if (resp.status === 200 && resp.data) {
         if (resp.data.token) {
+          dispatch('setDateTimeFormat', process.env.VUE_APP_DATE_FORMAT);
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:21]: We can have a fallback format when env file does not contain a date time format.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Settings.vue (hunk 20 above)
Comment (disha1202) [reply src/views/Settings.vue]: Made the requested change.

File: src/views/Settings.vue (hunk 22 above)
Comment (disha1202) [reply src/views/Settings.vue]: Updated url

File: src/views/Settings.vue (hunk 21 above)
Comment (disha1202) [reply src/views/Settings.vue]: Made the requested change.

File: src/store/modules/user/actions.ts (hunk 23 above)
Comment (disha1202) [reply src/store/modules/user/actions.ts]: Added fallback format

File: src/views/OrderDetail.vue [hunk 24]
```diff
@@ -213,6 +213,10 @@ export default defineComponent({
    this.fetchFacilities();
   },
   methods: {
+    isDateInvalid(){
+      const isInvalid = !this.ordersList.items.some((item: any) => DateTime.fromFormat(item.arrivalDate, this.dateTimeFormat).isValid);
```
Comment (ymaheshwari1) [inline src/views/OrderDetail.vue:217]:
```suggestion
      return !this.ordersList.items.some((item: any) => DateTime.fromFormat(item.arrivalDate, this.dateTimeFormat).isValid);
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/OrderDetail.vue (hunk 24 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Made the requested change.

**Review by ymaheshwari1 — APPROVED** [review]
Add comments when formatting the date time using luxon

**Review by dt2patel — APPROVED** [review]

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/UserState.ts [hunk 25]
```diff
@@ -3,4 +3,5 @@ export default interface UserState {
     current: object | null;
     currentFacility: object;
     instanceUrl: string;
+    dateTimeFormat: string;
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/store/modules/user/UserState.ts]: Updated variable name.

File: src/store/modules/user/actions.ts [hunk 26]
```diff
@@ -18,6 +18,7 @@ const actions: ActionTree<UserState, RootState> = {
       const resp = await UserService.login(username, password)
       if (resp.status === 200 && resp.data) {
         if (resp.data.token) {
+          dispatch('setDateTimeFormat', process.env.VUE_APP_DATE_FORMAT ? process.env.VUE_APP_DATE_FORMAT : 'MM/dd/yyyy');
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/store/modules/user/actions.ts]: Made the requested change.

File: src/views/OrderDetail.vue [hunk 27]
```diff
@@ -212,6 +213,10 @@ export default defineComponent({
    this.fetchFacilities();
   },
   methods: {
+    isDateInvalid(){
+      // Checked if any of the date format is different than the selected format.
+      return !this.ordersList.items.some((item: any) => DateTime.fromFormat(item.arrivalDate, this.dateTimeFormat).isValid);
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/views/OrderDetail.vue]: Made the requested change.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**